### ISSUE TITLE
Add Mg II 2800 angstrom doublet absorption feature to the list

### ISSUE THREAD

**Issue by biprateep** [desihub/prospect#27]

**Comment by biprateep** [comment]
Its in the Emission list but can also be seen in absorption

**Comment by armengau** [comment]
Let's use this open issue to summarize mail exchanges on lines.

- Lines are displayed in vaccum; no heliocentric frame correction to my knowledge.

- Suggestion for "major" lines (Tamara):
Emission: Ly-alpha, CIV, CIII, MgII, OII, Hdelta, Hgamma, Hbeta, OIII, OIII, Halpha, NII, SII, SII
Absorption: K, H, G, Mg, Na
Jeff suggests also to add for absorption H beta, gamma, delta, epsilon and zeta

- Lines to add (Alastair): UV absorption lines, Mg II as a doublet, NeV, ArIII at 9076A. Here is Alastair's detailled list (vaccum in principle):
Missing emission lines
3346.79   NeV   em
3426.85   NeV   em
9071.1    SIII  em
Missing optical absorption lines
8500.36   CaII  abs 
8544.44   CaII  abs triplet
8664.52   CaII  abs
IS low ionisation in UV 
SiII    1260.4221 abs
OI      1302.1685 abs
SiII    1304.3702 abs
CII     1334.5323 abs
SiIV  	1393.7602 abs
SiIV    1402.7728 abs
SV      1501.76   abs
SiII    1526.7070 abs
CIV     1548.204  abs
FeII    1608.4509 abs
AlII    1670.788  abs
SiII    1808.0129 abs
AlIII   1854.7183 abs
AlIII   1862.7911 abs
MgI     2026.4768 abs
FeII    2344.2130 abs
FeII    2374.4603 abs
FeII    2382.7642 abs
FeII    2389 2396 abs
emission lines in UV 
SiII*   1264.738  em
SiII*   1309.276  em
SiII*   1533.431  em
OIII]   1660.809  em
OIII]   1666.150  em
SiIII]  1882.468  em
SiIII]  1892.030  em
CIII]   1907.640  em
CIII]   1908.7346 em
both absorption and emission
MgII    2796.3543 abs and em
MgII    2803.5315 abs and em


- For record here is the current list of lines in prospect:
```
_line_list = [
    #
    # This is the set of emission lines from the spZline files.
    # See $IDLSPEC2D_DIR/etc/emlines.par
    # Wavelengths are in air for lambda > 2000, vacuum for lambda < 2000.
    # TODO: convert to vacuum wavelengths
    #
    {"name" : "Lyα",      "longname" : "Lyman α",        "lambda" : 1215.67,  "emission": True, "major": True  },
    {"name" : "Lyβ",      "longname" : "Lyman β",        "lambda" : 1025.18,  "emission": True, "major": False },
    {"name" : "N V",      "longname" : "N V 1240",       "lambda" : 1240.81,  "emission": True, "major": False },
    {"name" : "Si IV",    "longname" : "Si IV 1394",     "lambda" : 1393.75,  "emission": True, "major": False },
    {"name" : "Si IV",    "longname" : "Si IV 1403",     "lambda" : 1402.77,  "emission": True, "major": False },
    {"name" : "C IV",     "longname" : "C IV 1549",      "lambda" : 1549.48,  "emission": True, "major": True  },
    {"name" : "He II",    "longname" : "He II 1640",     "lambda" : 1640.42,  "emission": True, "major": False },
    {"name" : "C III]",   "longname" : "C III] 1908",    "lambda" : 1908.734, "emission": True, "major": False },
    {"name" : "Mg II",    "longname" : "Mg II 2799",     "lambda" : 2799.49,  "emission": True, "major": False },
    {"name" : "[O II]",   "longname" : "[O II] 3725",    "lambda" : 3726.032, "emission": True, "major": True  },
    {"name" : "[O II]",   "longname" : "[O II] 3727",    "lambda" : 3728.815, "emission": True, "major": True  },
    {"name" : "[Ne III]", "longname" : "[Ne III] 3868",  "lambda" : 3868.76,  "emission": True, "major": False },
    {"name" : "Hζ",       "longname" : "Balmer ζ",       "lambda" : 3889.049, "emission": True, "major": False },
    {"name" : "[Ne III]", "longname" : "[Ne III] 3970",  "lambda" : 3970.00,  "emission": True, "major": False },
    {"name" : "Hε",       "longname" : "Balmer ε",       "lambda" : 3970.072, "emission": True, "major": False },
    {"name" : "Hδ",       "longname" : "Balmer δ",       "lambda" : 4101.734, "emission": True, "major": False },
    {"name" : "Hγ",       "longname" : "Balmer γ",       "lambda" : 4340.464, "emission": True, "major": False },
    {"name" : "[O III]",  "longname" : "[O III] 4363",   "lambda" : 4363.209, "emission": True, "major": False },
    {"name" : "He II",    "longname" : "He II 4685",     "lambda" : 4685.68,  "emission": True, "major": False },
    {"name" : "Hβ",       "longname" : "Balmer β",       "lambda" : 4861.325, "emission": True, "major": False },
    {"name" : "[O III]",  "longname" : "[O III] 4959",   "lambda" : 4958.911, "emission": True, "major": True },
    {"name" : "[O III]",  "longname" : "[O III] 5007",   "lambda" : 5006.843, "emission": True, "major": True  },
    {"name" : "He II",    "longname" : "He II 5411",     "lambda" : 5411.52,  "emission": True, "major": False },
    {"name" : "[O I]",    "longname" : "[O I] 5577",     "lambda" : 5577.339, "emission": True, "major": False },
    {"name" : "[N II]",   "longname" : "[N II] 5755",    "lambda" : 5754.59,  "emission": True, "major": False },
    {"name" : "He I",     "longname" : "He I 5876",      "lambda" : 5875.68,  "emission": True, "major": False },
    {"name" : "[O I]",    "longname" : "[O I] 6300",     "lambda" : 6300.304, "emission": True, "major": False },
    {"name" : "[S III]",  "longname" : "[S III] 6312",   "lambda" : 6312.06,  "emission": True, "major": False },
    {"name" : "[O I]",    "longname" : "[O I] 6363",     "lambda" : 6363.776, "emission": True, "major": False },
    {"name" : "[N II]",   "longname" : "[N II] 6548",    "lambda" : 6548.05,  "emission": True, "major": False },
    {"name" : "Hα",       "longname" : "Balmer α",       "lambda" : 6562.801, "emission": True, "major": True  },
    {"name" : "[N II]",   "longname" : "[N II] 6583",    "lambda" : 6583.45,  "emission": True, "major": False },
    {"name" : "[S II]",   "longname" : "[S II] 6716",    "lambda" : 6716.44,  "emission": True, "major": False },
    {"name" : "[S II]",   "longname" : "[S II] 6730",    "lambda" : 6730.82,  "emission": True, "major": False },
    {"name" : "[Ar III]", "longname" : "[Ar III] 7135",  "lambda" : 7135.790, "emission": True, "major": False },
    #
    # Absorption lines
    #
    {"name" : "Hζ",   "longname" : "Balmer ζ",         "lambda" : 3889.049, "emission": False, "major": False },
    {"name" : "K",    "longname" : "K (Ca II 3933)",   "lambda" : 3933.7,   "emission": False, "major": False },
    {"name" : "H",    "longname" : "H (Ca II 3968)",   "lambda" : 3968.5,   "emission": False, "major": False },
    {"name" : "Hε",   "longname" : "Balmer ε",         "lambda" : 3970.072, "emission": False, "major": True },
    {"name" : "Hδ",   "longname" : "Balmer δ",         "lambda" : 4101.734, "emission": False, "major": False },
    {"name" : "G",    "longname" : "G (Ca I 4307)",    "lambda" : 4307.74,  "emission": False, "major": True },
    {"name" : "Hγ",   "longname" : "Balmer γ",         "lambda" : 4340.464, "emission": False, "major": False },
    {"name" : "Hβ",   "longname" : "Balmer β",         "lambda" : 4861.325, "emission": False, "major": False },
    {"name" : "Mg I", "longname" : "Mg I 5175",        "lambda" : 5175.0,   "emission": False, "major": False },
    {"name" : "D2",   "longname" : "D2 (Na I 5889)",   "lambda" : 5889.95,  "emission": False, "major": True },
    # {"name" : "D",    "longname" : "D (Na I doublet)", "lambda": 5892.9,   "emission": False, "major": False },
    {"name" : "D1",   "longname" : "D1 (Na I 5895)",   "lambda" : 5895.92,  "emission": False, "major": True },
    {"name" : "Hα",   "longname" : "Balmer α",         "lambda" : 6562.801, "emission": False, "major": True },
  ]
```
The lines are vaccum converted, using:
```
def _airtovac(w):
    """Convert air wavelengths to vacuum wavelengths. Don't convert less than 2000 Å.
    Parameters
    ----------
    w : :class:`float`
        Wavelength [Å] of the line in air.
    Returns
    -------
    :class:`float`
        Wavelength [Å] of the line in vacuum.
    """
    if w < 2000.0:
        return w;
    vac = w
    for iter in range(2):
        sigma2 = (1.0e4/vac)*(1.0e4/vac)
        fact = 1.0 + 5.792105e-2/(238.0185 - sigma2) + 1.67917e-3/(57.362 - sigma2)
        vac = w*fact
    return vac
```

**Comment by moustakas** [comment]
Of possible utility / interest: the line-lists used for some of the spectral sims / templates are here:
* https://github.com/desihub/desisim/blob/master/py/desisim/data/recombination_lines.ecsv
* https://github.com/desihub/desisim/blob/master/py/desisim/data/forbidden_lines.ecsv

**Comment by armengau** [comment]
The line list was updated (in my current dev branch, disp-models):
- "Major" lines as proposed by Tamara (both emission and absorption).
- Included all of entries in Alastair's list. When an entry already existed in SDSS list, I kept SDSS entry.
- Checked all entries from desisim lists (recombination and forbidden lines): added lines whose quoted line-ratio wrt Hbeta is > 0.1: H8, HeI_3389 ; and ArIII]_7751.
